src/monattack.cpp

```diff
@@ -4367,6 +4370,53 @@ bool mattack::darkman( monster *z )
     return true;
 }
 
+bool mattack::shadow_hunter( monster *z )
+{
+    bool damaged = z->has_effect( effect_took_damage );
+    Creature *target = z->attack_target();
+    if( target == nullptr ) {
+        return false;
+    }
+    //Hide and teleport randomly after being attacked or just randomly
+    if( ( damaged && one_in( 3 ) ) || ( one_in( 15 ) ) ) {
+        int tries = 0;
+        int maxtries = 20;
+        int telerange = 30;
+        tripoint newpos = z->pos();
+        do {
+            newpos.x = rng( z->posx() - telerange, z->posx() + telerange );
+            newpos.y = rng( z->posy() - telerange, z->posy() + telerange );
+            tries++;
+        } while( ( tries > maxtries ) &&
```

Comment:
`tries > maxtries` will never be true as `tries` is 1 when this is evaluated for the first time. And that cancels the loop, so it's never incremented further. I assume you meant `tries < maxtries` here.